### PR TITLE
Fix/dev mode behavior when Jetpack is connected 

### LINKED ISSUE #2156: Dev Mode: Jetpack->Settings acts connected
Originally reported https://github.com/Automattic/jetpack/issues/2153#issuecomment-104127907

If a site is connected to WP.com and then activated Dev Mode, Jetpack->Settings will still act fully connected, including the Disconnect from WordPress.com link at the bottom.

For the staging sites (see #2153, etc), the cloned db will believe it is connected, so Development Mode should override that as much as possible. In a staging site scenario, disconnecting from WP.com will terminate the production site (as the token destroyed is from prod).

### PR DESCRIPTION
**Pull request by dereksmart**
Make dev mode act like dev mode even when it was turned on while Jetpack was connected.  

Here's a list of what will now happen when development mode is activated on a connected Jetpack site.  
- Hides "disconnect from WordPress" link
- Kills any sync communication to dotcom
- Greys out `requires_connection` modules on the settings page
- Can't edit settings for those modules either, nor activate/deactivate them

For more context, check out #2156 

Fixes #2156 

cc @kraftbj

### CODE REVIEW

File: _inc/footer.php [hunk 1]
```diff
@@ -28,7 +28,7 @@
 					<a href="http://jetpack.me/contact-support/" title="<?php esc_attr_e( 'Contact the Jetpack Happiness Squad.', 'jetpack' ); ?>"><?php _e( 'Support', 'jetpack' ); ?></a>
 					<a href="http://jetpack.me/survey/?rel=<?php echo JETPACK__VERSION; ?>" title="<?php esc_attr_e( 'Take a survey.  Tell us how we&#8217;re doing.', 'jetpack' ); ?>"><?php _e( 'Give Us Feedback', 'jetpack' ); ?></a>
 
-					<?php if ( $is_active && current_user_can( 'jetpack_disconnect' ) ) : ?>
+					<?php if ( $is_active && current_user_can( 'jetpack_disconnect' ) && ! Jetpack::is_development_mode() ) : ?>
```
Comment (georgestephanis) [inline _inc/footer.php]: Wouldn't this be unnecessary if we modified $is_active so it's never showing as active in dev mode?

Comment (dereksmart) [inline _inc/footer.php]: Yes it would

File: _inc/footer.php [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <?php
 global $current_user;
-$is_active         = Jetpack::is_active();
+$is_active         = Jetpack::is_active() && ! Jetpack::is_development_mode();
 $user_token        = Jetpack_Data::get_access_token( $current_user->ID );
```
Comment (roccotripaldi) [inline _inc/footer.php]: You may want to group the dependent conditions in parenthesis.

**Comment by roccotripaldi** [comment]
Code looks solid, will try to test this evening.

**Comment by georgestephanis** [comment]
Just confirming, this means that ones that require connection but are active will stay in the active option, but never get included, yes?

**Comment by dereksmart** [comment]
@georgestephanis correct.  Options/settings are preserved for mods that require connection.

**Comment by zinigor** [comment]
Read the changes, tested, everything works as expected!

**Pull request merged**